src/platform/PICO/io_def_generated.h

```diff
@@ -24,195 +24,193 @@
 // DEFIO_PORT_<port>_USED_COUNT is count of used pins on target
 
 #if defined(RP2350A)
-#define DEFIO_PORT_A_USED_COUNT 30
+#define DEFIO_USED_COUNT 30
 #elif defined(RP2350B)
-#define DEFIO_PORT_A_USED_COUNT 48
+#define DEFIO_USED_COUNT 48
 #else
 #error "Unsupported target MCU type for PICO"
 #endif
 
-#define DEFIO_PORT_A_OFFSET (0)
-
-// DEFIO_GPIOID__<port> maps to port index
-#define DEFIO_GPIOID__A 0
+// DEFIO_GPIOID__ --> there is only one gpioid
+#define DEFIO_GPIOID__ 0
 
 // DEFIO_TAG__P<port><pin> will expand to TAG if defined for target, error is triggered otherwise
 // DEFIO_TAG_E__P<port><pin> will expand to TAG if defined, to NONE otherwise (usefull for tables that are CPU-specific)
 // DEFIO_REC__P<port><pin> will expand to ioRec* (using DEFIO_REC_INDEX(idx))
 
-#define DEFIO_TAG__PA0    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 0)
-#define DEFIO_TAG__PA1    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 1)
-#define DEFIO_TAG__PA2    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 2)
-#define DEFIO_TAG__PA3    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 3)
-#define DEFIO_TAG__PA4    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 4)
-#define DEFIO_TAG__PA5    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 5)
-#define DEFIO_TAG__PA6    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 6)
-#define DEFIO_TAG__PA7    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 7)
-#define DEFIO_TAG__PA8    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 8)
-#define DEFIO_TAG__PA9    DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 9)
-#define DEFIO_TAG__PA10   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 10)
-#define DEFIO_TAG__PA11   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 11)
-#define DEFIO_TAG__PA12   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 12)
-#define DEFIO_TAG__PA13   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 13)
-#define DEFIO_TAG__PA14   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 14)
-#define DEFIO_TAG__PA15   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 15)
-#define DEFIO_TAG__PA16   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 16)
-#define DEFIO_TAG__PA17   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 17)
-#define DEFIO_TAG__PA18   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 18)
-#define DEFIO_TAG__PA19   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 19)
-#define DEFIO_TAG__PA20   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 20)
-#define DEFIO_TAG__PA21   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 21)
-#define DEFIO_TAG__PA22   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 22)
-#define DEFIO_TAG__PA23   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 23)
-#define DEFIO_TAG__PA24   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 24)
-#define DEFIO_TAG__PA25   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 25)
-#define DEFIO_TAG__PA26   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 26)
-#define DEFIO_TAG__PA27   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 27)
-#define DEFIO_TAG__PA28   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 28)
-#define DEFIO_TAG__PA29   DEFIO_TAG_MAKE(DEFIO_GPIOID__A, 29)
+#define DEFIO_TAG__P0    DEFIO_TAG_MAKE(DEFIO_GPIOID__, 0)
```

Comment:
Thi can be:
```suggestion
#define DEFIO_TAG__P0   DEFIO_TAG_MAKE(0)
```

IMO it is better to keep one level of macro indirection, it may be useful to expand to `(1)` so that some boost-pp tricks can be used with TAGs.